### DISCUSSION TITLE
vitepress do not work on host where the subpath is unknown

### DISCUSSION THREAD

**Discussion by wighawag** [vuejs/vitepress#3144] [General]
Hi everyone, posting this message as I did not get feedback on the following reproducible issue: https://github.com/vuejs/vitepress/issues/3057

Would love my website to work regardless of where it is hosted. Currently vitepress assume that basepath is fixed and known

As mentioned in the issue, there are several cases where this not the case:

- archive website like the [wayback machine](https://archive.org/) that host website on sub path
- [IPFS](https://ipfs.tech/) or other content-addressable network where the pathname is a hash that cannot be known before build time
- some website that let you embed other website but where you are not in control of the path

Would love to know if the vitepress team is planning to support this or not ?  And would help them to make it happen ?

**Comment by fvsch** [comment]
Is the solution you have in mind something like: "a mode where all links to pages and URLs of assets are relative"?

I’m wondering if that's compatible with VitePress's design. I know it can work with a static site generator that never uses JavaScript for navigation, but I believe VitePress may use JavaScript to take over navigation currently?

That can be an issue if your sidebar has links relative to the current page, say:

```html
<!-- current page is /docs/abc/def -->
<a href="../xyz">Go to XYZ</a>
```

Then if you click to navigate, and if it loads data for `/docs/xyz` and changes the current URL and the main page content to `/docs/xyz` but doesn't update the whole sidebar:

- links in the main content would be correct;
- links in the sidebar would be wrong (relative to the old URL).

But I’m not sure how VitePress handles navigation currently.

**Reply by brc-dd** [reply]
Just for context, some SSGs support specifying `./` as `base`. That generates the URLs you're wishing. But that's, honestly, too much of a refactor for us. Currently the base is constrained to `/<something>/` or `/`. I can start working on that if there is much demand, but that's gonna need few weeks dedicated time, which I'm not sure if I can commit to right now.

**Reply by wighawag** [reply]
@fvsch yes that how I expect it to work. svelte-kit, which uses vite supports that, even with javascript taking over the navigation once hydrated

@brc-dd Is there anything I coulddo  to help make it a possibility ? 

Or maybe pointer on alternative solution that would not require changes in vitepress except maybe some metadata that could be generated for a post-processing tool to update links. ?

I am mentioning a post-processing tool as this is how I worked arround svelte-kit until they supported the feature from the get-go
